### PR TITLE
[rush-lib] Use pnpm-sync-lib logging APIs to customize the log message for pnpm-sync operations.

### PR DESCRIPTION
**Pull request by g-chao**
## Summary
`pnpm-sync-lib` now have a logMessage API which allow us to customize the log message printing behavior. Check [here](https://github.com/tiktok/pnpm-sync/pull/15) to see how it implemented.

So, on Rush side, we can use this API to preventing unexpected log message and customize the log message format. 

## Details
Generally, there will be four types of message coming from `pnpm-sync-lib`, there are `'info' | 'warning' | 'error' | 'verbose'`.
1. Normally, we will only print out `info` message. 
2. `error` and `warning` will be printed if there are errors or warnings. 
3. The `verbose` will be printed out if we pass `--debug` or `--vebose` flag to the rush commands.

The `pnpm-sync-lib` also provides operation metadata in the API, so if don't want the message content from `pnpm-sync-lib`, we can utilize that metadata to fully customize the log message on the Rush side. 

## How it was tested
Manually tested on local.

## Impacted documentation
N/A

### CODE REVIEW

File: libraries/rush-lib/src/cli/RushXCommandLine.ts
Comment (iclanton) [inline libraries/rush-lib/src/cli/RushXCommandLine.ts:247]: Can you use the `Terminal` API in this file? You don't have to convert everything, but it'd be good to get that started.

File: libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts
Comment (iclanton) [inline libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:410]: Verbosity is handled by `Terminal`.

File: libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts
Comment (iclanton) [inline libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts:548]:
```suggestion
              this._terminal.writeVerboseLine(message);
```

File: libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts
Comment (iclanton) [inline libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts:551]: Arbitrary newline?

File: libraries/rush-lib/src/logic/operations/PnpmSyncCopyOperationPlugin.ts
Comment (iclanton) [inline libraries/rush-lib/src/logic/operations/PnpmSyncCopyOperationPlugin.ts:66]:
```suggestion
                    this._terminal.writeVerboseLine(message);
```

File: libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts
Comment (iclanton) [inline libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts:536]: Extract this to a shared function?

File: libraries/rush-lib/src/logic/operations/PnpmSyncCopyOperationPlugin.ts
Comment (iclanton) [inline libraries/rush-lib/src/logic/operations/PnpmSyncCopyOperationPlugin.ts:22]:
```suggestion
  private readonly _terminal: ITerminal;

  public constructor(terminal: ITerminal) {
    this._terminal = terminal;
  }
  
```

File: libraries/rush-lib/src/logic/operations/PnpmSyncCopyOperationPlugin.ts
Comment (iclanton) [inline libraries/rush-lib/src/logic/operations/PnpmSyncCopyOperationPlugin.ts:11]: Put this at the top with the other external dependencies.

**Review by iclanton — CHANGES_REQUESTED** [review]

**Comment by octogonz** [comment]
@g-chao I've proposed some upstream changes that should to go in first: https://github.com/tiktok/pnpm-sync/pull/20

File: libraries/rush-lib/src/utilities/PnpmSyncUtilities.ts
Comment (octogonz) [inline libraries/rush-lib/src/utilities/PnpmSyncUtilities.ts:33]: @g-chao  For some reason this doesn't get printed with `rush rebuild --verbose`.  I didn't have time to investigate why.

Comment (g-chao) [reply libraries/rush-lib/src/utilities/PnpmSyncUtilities.ts:33]: I checked a little bit, this seems a general bug. Basically, the `--verbose` is not take effect for `rush rebuild` command.

Comment (octogonz) [reply libraries/rush-lib/src/utilities/PnpmSyncUtilities.ts:33]: The top-level Rush terminal provider is created here:

https://github.com/microsoft/rushstack/blob/e0de053d02bc10128933cb8e75bb1657b636bb3c/libraries/rush-lib/src/cli/RushCommandLineParser.ts#L106-L107

It sets `verboseEnabled` based on the `--debug` flag not the `--verbose` flag:

https://github.com/microsoft/rushstack/blob/e0de053d02bc10128933cb8e75bb1657b636bb3c/libraries/rush-lib/src/cli/RushCommandLineParser.ts#L190-L193

I wonder if this was a historical decision? Maybe because of this definition of `--verbose`?
```
  -v, --verbose         Display the logs during the build, rather than just
                        displaying the build status summary
```
Or maybe because not all commands have the `--verbose` flag?

I'll discuss it with some other maintainers today.

@dmichon-msft

Comment (octogonz) [reply libraries/rush-lib/src/utilities/PnpmSyncUtilities.ts]: In afffe7c7820d5da1a80e0806da3114677a51604f I have wired up the terminal so that `pnpm-sync` output is now printed with `rush --debug`.  To unblock this PR, we'll address the problem where `--verbose` doesn't work in a followup PR.

**Pull request merged**
